### PR TITLE
add identity field pattern

### PR DESCRIPTION
**Pull request by lkpengcs**
In this pull request,

- Add the identity field pattern
- This is related to this [issue](https://github.com/iluwatar/java-design-patterns/issues/1298)

Pull request description

- Implement the identity field pattern according to the given [link](https://www.sourcecodeexamples.net/2018/05/identity-field-pattern.html).

> For detailed contributing instructions see https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

### CODE REVIEW

**Review by ohbus — CHANGES_REQUESTED** [review]
Thank you so much for your efforts!

Can you please check the 2 code smells reported by Sonar?

Would love to have a perfect submission with full coverage analysis as well.

**Comment by lkpengcs** [comment]
hi, I have removed the code smells. As for the coverage, I can have full coverage locally. Not sure about how Sonar checks this.

File: identity-field/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+layout: pattern
+title: Identity-Field
```
Comment (iluwatar) [inline identity-field/README.md:3]:
```suggestion
title: Identity Field
```

File: identity-field/pom.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.22</version>
```
Comment (iluwatar) [inline identity-field/pom.xml:28]: Please use the version from the parent pom.xml

File: identity-field/pom.xml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                <executions>
+                    <execution>
+                        <id>make-assembly</id>
+                        <phase>package</phase> <!-- packaging phase -->
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
```
Comment (iluwatar) [inline identity-field/pom.xml:54]: The configuration seems overly verbose. Can you check how maven-assembly-plugin has been configured in the other patterns?

File: identity-field/pom.xml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
+                    <target>8</target>
+                </configuration>
+            </plugin>
```
Comment (iluwatar) [inline identity-field/pom.xml:62]: This should be left out. Use the Java version that's coming from the parent pom.xml (11)

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Comment by iluwatar** [comment]
Closed due to inactivity

**Pull request closed without merging**